src/Nethermind/Nethermind.HealthChecks/ClHealthRequestsTracker.cs

```diff
@@ -0,0 +1,74 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.Api;
+using Nethermind.Core;
+using Nethermind.Logging;
+
+namespace Nethermind.HealthChecks;
+
+public class ClHealthRequestsTracker(ITimestamper timestamper, int maxIntervalClRequestTime, ILogger logger)
+    : IEngineRequestsTracker, IClHealthTracker
+{
+    private const int ClUnavailableReportMessageDelay = 5;
+
+    private DateTime _latestForkchoiceUpdated = timestamper.UtcNow;
+    private DateTime _latestNewPayload = timestamper.UtcNow;
+
+    private Timer _timer;
+
+    public Task StartAsync()
+    {
+        _timer = new Timer(ReportClStatus, null, TimeSpan.Zero,
+            TimeSpan.FromSeconds(ClUnavailableReportMessageDelay));
+
+        return Task.CompletedTask;
+    }
+
+    private Task StopAsync()
+    {
+        _timer.Change(Timeout.Infinite, 0);
+
+        return Task.CompletedTask;
+    }
+
+    public async ValueTask DisposeAsync()
+    {
+        await StopAsync();
+        await _timer.DisposeAsync();
+    }
+
+    private void ReportClStatus(object _)
+    {
+        if (!CheckClAlive())
+        {
+            if (logger.IsWarn)
+                logger.Warn("Not receiving ForkChoices from the consensus client that are required to sync.");
+        }
+    }
+
+    private bool IsRequestTooOld(DateTime now, DateTime requestTime)
+    {
+        int diff = (int)(Math.Floor((now - requestTime).TotalSeconds));
+        return diff > maxIntervalClRequestTime;
```

Comment:
Prefer to use `TimeSpan` over `int` for more control over durations (requires changing the `maxIntervalClRequestTime` parameter type too):

```suggestion
        TimeSpan diff = (now - requestTime).Duration();
        return diff > maxIntervalClRequestTime;
```

Reply:
We cant change it's type because of backwards compatibility

Reply:
Without changing the `maxIntervalClRequestTime` parameter type we can do the following:

```suggestion
        TimeSpan diff = (now - requestTime).Duration();
        return diff > TimeSpan.FromSeconds(maxIntervalClRequestTime);
```